modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-core/src/main/java/org/nuxeo/apidoc/snapshot/SnapshotManagerComponent.java

```diff
@@ -104,6 +100,9 @@
 
     @Override
     public DistributionSnapshot getRuntimeSnapshot() {
+        if (isSiteMode()) {
+            throw new RuntimeException("Live runtime cannot be snapshotted.");
```

Comment:
Could it be interesting to throw a `NuxeoException` with an HTTP bad request status?

Reply:
I indeed hesitated with NuxeoException, but it's from core-api, and this is not really a core issue. I would have picked a (Nuxeo)RuntimeException if there was one, otherwise maybe it makes sense to create a specific ApidocException, wdyt?
And this is low level now, so i don't see why we'd want to link this to HTTP unless that's a common practice in that kind of use case (?)

Reply:
We use `NuxeoException` for our generic runtime exception if it's the available dependencies. In the pure runtime, we have our `RuntimeServiceException`

Reply:
I thought this method could be called on an HTTP request if it's not the case no need to have an HTTP status indeed.

The `ApidocException` could be interesting if we want to do something with it. If it's just about not using the generic `RuntimeException`, the `RuntimeServiceException` should be fine.